java/src/main/java/org/vowpalwabbit/spark/VowpalWabbitExample.java

```diff
@@ -33,19 +33,34 @@
      * {@code baseIndex} is expected to be pre-hashed (e.g. hash(namespace)).
      *
      * @param ns        the first character of the namespace.
-     * @param baseIndex the base index for each of the {@code values}.
+     * @param baseIndex the base index for each of the {@code values}. This is also the namespace hash (e.g. hash(namespace))
      * @param values    the feature values.
      */
     public native void addToNamespaceDense(char ns, int baseIndex, double[] values);
 
     /**
-     * Adds the sparse features values to the supplied naemspace.
+     * Adds the sparse features values to the supplied namespace.
      *
      * @param ns      the first character of the namespace.
      * @param indices the indices of each corresponding feature value.
      * @param values  the feature values.
+     * @deprecated use the version with baseIndex instead
      */
-    public native void addToNamespaceSparse(char ns, int[] indices, double[] values);
+    @Deprecated
+    public void addToNamespaceSparse(char ns, int[] indices, double[] values)
+    {
+        addToNamespaceSparse(ns, ns, indices, values);
+    }
+
+        /**
+     * Adds the sparse features values to the supplied namespace.
+     *
+     * @param ns      the first character of the namespace.
+     * @param baseIndex the namespace hash (e.g. hash(namespace))
+     * @param indices the indices of each corresponding feature value.
+     * @param values  the feature values.
+     */
+    public native void addToNamespaceSparse(char ns, int baseIndex, int[] indices, double[] values);

```

Comment:
Wondering if there should be another addToNamespace overload that takes the whole namespace string.

Reply:
This is simply enabling the spark java interface to work with this. The spark java interface requires manual hashing of your own namespace and was an intentional decision when it was created for performance reasons.